Source/WTF/wtf/HashTable.h

```diff
@@ -631,44 +631,35 @@ DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(HashTable);
     {
     }
 
-#if !ASSERT_ENABLED
-
-    template<typename Key, typename Value, typename Extractor, typename HashFunctions, typename Traits, typename KeyTraits>
+    template<typename Key, typename Value, typename Extractor, typename HashFunctions, typename Traits, typename KeyTraits, ShouldValidateKey validateKey>
     template<typename HashTranslator, typename T>
-    inline void HashTable<Key, Value, Extractor, HashFunctions, Traits, KeyTraits>::checkKey(const T&)
+    void HashTable<Key, Value, Extractor, HashFunctions, Traits, KeyTraits, validateKey>::checkKey(const T& key)
     {
-    }
-
-#else // ASSERT_ENABLED
+        if constexpr (validateKey == ShouldValidateKey::No)
+            return;
 
-    template<typename Key, typename Value, typename Extractor, typename HashFunctions, typename Traits, typename KeyTraits>
-    template<typename HashTranslator, typename T>
-    void HashTable<Key, Value, Extractor, HashFunctions, Traits, KeyTraits>::checkKey(const T& key)
-    {
         if (!HashFunctions::safeToCompareToEmptyOrDeleted)
             return;
-        ASSERT(!HashTranslator::equal(KeyTraits::emptyValue(), key));
+        RELEASE_ASSERT(!HashTranslator::equal(KeyTraits::emptyValue(), key));
```

Comment:
Wait, this would mean we no longer validate keys in debug builds for regular hash maps.
I don't think we want that behavior change.

Reply:
I believe 'RELEASE_ASSERT()' asserts on both Debug and Release builds, correct?

Reply:
Yes but you're using ShouldValidateKey::No for regular hash maps so these release asserts would never run.

Reply:
Ah, got it! I'll add in a conditional guard that will force debug builds to validateKey and assert on invalid keys (similar to how behavior is before this change). Do you suggest a different approach?